### ISSUE TITLE
{"error": "invalid_client"}

### ISSUE THREAD

**Issue by lucacorti** [jazzband/django-oauth-toolkit#52]
Hello,

I'm following the tutorial for Django Rest Framework. I set up my application in the admin and I'm sending a POST request just like in the tutorial, but always get

```
{"error": "invalid_client"}
```

as a response:

```
$ curl -vK user -X POST -d "grant_type=password&username=myuser&password=mypass" http://localhost:8000/o/token/
* About to connect() to localhost port 8000 (#0)
*   Trying 127.0.0.1...
* connected
* Connected to localhost (127.0.0.1) port 8000 (#0)
* Server auth using Basic with user 'RgmI&_Y A5mdjuAh2T/]m_ZeD|u'[8`$r|`k'!'
> POST /o/token/ HTTP/1.1
> Authorization: Basic UmdtSSZfWSBBNW1kanVBaDJUL11tX1plRHx1J1s4YCRyfGBrJyE6dnZdWWw/LUB3YWNBI2Q+NEcyNWEkbkZCclogJGQ/e1ZtYm4rMl5GYnRNSl9ZTitiVHB6b3UjXkRdaFhMeGxNL28jJz95MCV0ZXQ8Z3w8OEZTZHVPMkIvN3Y6dG5EUz99ZGxla3B2dTchWTssJDA8VEQ1UmwucA==
> User-Agent: curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8x zlib/1.2.5
> Host: localhost:8000
> Accept: */*
> Content-Length: 55
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 55 out of 55 bytes
* HTTP 1.0, assume close after body
< HTTP/1.0 400 BAD REQUEST
< Date: Sun, 08 Sep 2013 17:30:37 GMT
< Server: WSGIServer/0.1 Python/2.7.5
< Content-Language: it
< Vary: Accept-Language, Cookie
< Pragma: no-cache
< Cache-Control: no-store
< X-Frame-Options: SAMEORIGIN
< Content-Type: application/json;charset=UTF-8
< 
* Closing connection #0
{"error": "invalid_client"}
```

**Comment by fgau** [comment]
@twolff-iow thanks for your comment, I had the same problem on my apache. problem solved with "WSGIPassAuthorization On". thank you!

**Comment by bquinn** [comment]
@twolff-iow that caught me as well, thanks for the tip! It would be great if that was added to the documentation somewhere.

**Comment by RaulBejarano** [comment]
@ghost that solved my problem, thanks!

**Comment by lesserenterpriser** [comment]
@ghost, thanks for the comment.  Solved my issue.

**Comment by n2ygk** [comment]
@prahlad1989 Please take a look at #1161 to see if this helps. There are two breaking changes with release 2.0.0:
- PKCE is now required
- client_secret is now hashed so you have to get the secret before it is saved

Some fixes and documentation improvements have been merged into the master branch which will help once released. Just waiting on a couple of PRs to be merged.
